test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -1022,14 +1016,18 @@ func TestPrebindPlugin(t *testing.T) {
 				t.Errorf("Error while creating a test pod: %v", err)
 			}
 
-			if test.fail || test.reject {
+			if test.fail {
 				if test.succeedOnRetry {
 					if err = testutils.WaitForPodToScheduleWithTimeout(testCtx.ClientSet, pod, 10*time.Second); err != nil {
 						t.Errorf("Expected the pod to be schedulable on retry, but got an error: %v", err)
 					}
 				} else if err = wait.Poll(10*time.Millisecond, 30*time.Second, podSchedulingError(testCtx.ClientSet, pod.Namespace, pod.Name)); err != nil {
 					t.Errorf("Expected a scheduling error, but didn't get it. error: %v", err)
 				}
+			} else if test.reject {
+				if err = testutils.WaitForPodToScheduleWithTimeout(testCtx.ClientSet, pod, 5*time.Second); err == nil {
```

Comment:
Can you use WaitForPodUnschedulable instead?

Reply:
`WaitForPodUnschedulable` also call `WaitForPodToScheduleWithTimeout` underlying, and the timeout will be 30s, it's too long I think.

Reply:
it calls `WaitForPodUnschedulableWithTimeout`, which relies on the Pod condition to show up in the Pod status.

Reply:
Updated